strawberry/schema/execute.py

```diff
@@ -256,3 +259,135 @@ def execute_sync(
+                )
+                return  # pragma: no cover
+
+        async with extensions_runner.validation():
+            _run_validation(execution_context)
+            if execution_context.errors:
+                process_errors(execution_context.errors, execution_context)
+                yield False, ExecutionResult(data=None, errors=execution_context.errors)
+                return  # pragma: no cover
+
+        async def process_result(result: GraphQLExecutionResult):
+            execution_context.result = result
+            # Also set errors on the execution_context so that it's easier
+            # to access in extensions
+            if result.errors:
+                execution_context.errors = result.errors
+
+                # Run the `Schema.process_errors` function here before
+                # extensions have a chance to modify them (see the MaskErrors
+                # extension). That way we can log the original errors but
+                # only return a sanitised version to the client.
+                process_errors(result.errors, execution_context)
+
+            return ExecutionResult(
+                data=execution_context.result.data,
+                errors=execution_context.result.errors,
+                extensions=await extensions_runner.get_extensions_results(),
+            )
+
+        async with extensions_runner.executing():
+            # currently original_subscribe is an async function.  A future release
+            # of graphql-core will make it optionally awaitable
+            result: Union[AsyncIterable[GraphQLExecutionResult], GraphQLExecutionResult]
+            result_or_awaitable = original_subscribe(
+                schema,
+                execution_context.graphql_document,
+                root_value=execution_context.root_value,
+                context_value=execution_context.context,
+                variable_values=execution_context.variables,
+                operation_name=execution_context.operation_name,
+            )
+            if isawaitable(result_or_awaitable):
+                result = await cast(
+                    Awaitable[
+                        Union[
+                            AsyncIterable["GraphQLExecutionResult"],
+                            "GraphQLExecutionResult",
+                        ]
+                    ],
+                    result_or_awaitable,
+                )
+            else:  # pragma: no cover
+                result = cast(
+                    Union[
+                        AsyncIterable["GraphQLExecutionResult"],
+                        "GraphQLExecutionResult",
+                    ],
+                    result_or_awaitable,
+                )
+
+            if isinstance(result, GraphQLExecutionResult):
+                yield False, await process_result(result)
+                return
+
+            aiterator = result.__aiter__()
+            try:
```

Comment:
why not `contextlib.suppress`?

Reply:
I don't understand.
> Because that exception might come from anywhere

What's wrong with
  
```py
with contextlib.supress(BaseException):
    async for result in aiterator:
        yield True, await process_result(result)
finally:
if hasattr(aiterator, "aclose"):
    await aiterator.aclose()
```

Reply:
No, you don't want to suppress BaseException.  You never want to do that.  CancelledError is one reason.
No, try-finally exists precisely do do this kind of thing.  In fact, that is how `contextlib.aclosing()` is implemented and it would be appropriate there, except that a) it requires python 3.8 and b), it _assumes_ that `aclose()` is present.